model/openapi.yml

```diff
@@ -674,7 +756,11 @@ components:
         events:
           type: array
           items:
-            $ref: "#/components/schemas/event"
+            anyOf:
+              - $ref: "#/components/schemas/genericEvent"
+              - $ref: "#/components/schemas/upgradeEvent"
+              - $ref: "#/components/schemas/diagnosticsEvent"
+              - $ref: "#/components/schemas/inputEvent"
```

Comment:
Event acks do not have a specific type per action as most actions use `genericEvent` with an optional error for most actions.
I don't think we need to have a 1-1 action-type to ack-event mapping as the agent only produces different acks for upgrades (may include an error and payload, but the upgrade communication is being replaced by upgrade_details in the checkin), diagnostics (may include an error and data), and the input action (lots of additional fields); but i'm open to feedback here.

We also can't use a discriminator here as action/ack types are undefined in the object/parent (type info is read from the separate action record)